### ISSUE TITLE
Motif finding documentation

### ISSUE THREAD

**Issue by mavencode01** [graphframes/graphframes#150]
Is there any documentation to explain in details how to write complex queries with motif finding DSL? 

I want to write a DSL that shows if a node is connected to another node at any n-level but don't know
how to express this with the limited documentation on motif.

    (a)-[]-...->(b)

Any idea how to go about this?

**Comment by felixcheung** [comment]
hmm, there's a bit on variable length motif at the end of this section:
http://graphframes.github.io/user-guide.html#motif-finding

**Comment by mavencode01** [comment]
@felixcheung Is there any example to query a variable length motif ?

```
Currently, in order to search for variable-length motifs, users need to run one query for each    
possible length. However, the above query patterns allow users to re-use the same code for each 
length, with the only change being to update the sequence of motif elements (“ab”, “bc”, “cd” 
above).
```

**Comment by felixcheung** [comment]
I thought maybe there was but after some searches I'm not finding examples of it...

**Comment by thunterdb** [comment]
The example that @felixcheung mentioned does that but it may be a bit hard to follow:
 - you build the longest chain that you care about using the motif pattern
 - you incrementally build the chain up to the maximum, keeping track of its length using an accumulator
 - then you can filter out the lengths that are relevant

It is probably easier (but not as efficient) to start by running repeated queries, one for each length, and then optimize with a fold.

**Comment by pgok** [comment]
@thunterdb Can you please share some (pseudo?) code to do this? I am trying to do the same thing as @mavencode01

**Comment by akshayagarwalgithub** [comment]
Hi @felixcheung, @thunterdb 
                           I am using motif finding for my project, but seeing lot of shuffling happening for the simple chain I have built including some filter operation on vertex and edges. Is there any documentation or if you can throw some light on how the motif finding internally works doing graph traversal, that it is leading to so many shuffles.

**Comment by felixcheung** [comment]
@akshayagarwalgithub actually this is somewhat to be expected as you can see, motif finding is implemented as a sequence of joins: https://github.com/graphframes/graphframes/blob/master/src/main/scala/org/graphframes/GraphFrame.scala#L816

I think that it would be good to add some doc on how it works and best practices - all those with joining would apply here.

**Comment by rjurney** [comment]
I have just added a lengthy motif finding tutorial we are about to publish :)
